src/lib/opcode-labels.js

```diff
@@ -1,126 +1,248 @@
+     */
+    setTranslatorFunction (translator) {
+        this._translator = translator;
+        this._refreshOpcodeMap();
+    }
+
+    /**
+     * Internal function to update opcode Map when translation function is defined
+     * @private
+     */
+    _refreshOpcodeMap () {
+        // Motion
+        this._opcodeMap.motion_direction.labelFn = () => this._translator(messages.motion_direction);
+        this._opcodeMap.motion_xposition.labelFn = () => this._translator(messages.motion_xposition);
+        this._opcodeMap.motion_yposition.labelFn = () => this._translator(messages.motion_yposition);
+
+        // Looks
+        this._opcodeMap.looks_size.labelFn = () => this._translator(messages.looks_size);
+        this._opcodeMap.looks_costumenumbername.labelFn = params => {
+            if (params.NUMBER_NAME === 'number') {
+                return this._translator(messages.looks_costumenumber);
+            }
+            return this._translator(messages.looks_costumename);
+        };
+        this._opcodeMap.looks_backdropnumbername.labelFn = params => {
+            if (params.NUMBER_NAME === 'number') {
+                return this._translator(messages.looks_backdropnumber);
+            }
+            return this._translator(messages.looks_backdropname);
+        };
+        this._opcodeMap.looks_backdropname.labelFn = () => this._translator(messages.looks_backdropname);
+
+        // Data
+        this._opcodeMap.data_variable.labelFn = params => params.VARIABLE;
+        this._opcodeMap.data_listcontents.labelFn = params => params.LIST;
+
+        // Sound
+        this._opcodeMap.sound_volume.labelFn = () => this._translator(messages.sound_volume);
+        this._opcodeMap.sound_tempo.labelFn = () => this._translator(messages.sound_tempo);
+
+        // Sensing
+        this._opcodeMap.sensing_answer.labelFn = () => this._translator(messages.sensing_answer);
+        this._opcodeMap.sensing_loudness.labelFn = () => this._translator(messages.sensing_loudness);
+        this._opcodeMap.sensing_username.labelFn = () => this._translator(messages.sensing_username);
+        this._opcodeMap.sensing_current.labelFn = params => {
+            switch (params.CURRENTMENU) {
+            case 'year':
+                return this._translator(messages.sensing_current_year);
+            case 'month':
+                return this._translator(messages.sensing_current_month);
+            case 'date':
+                return this._translator(messages.sensing_current_date);
+            case 'dayofweek':
+                return this._translator(messages.sensing_current_dayofweek);
+            case 'hour':
+                return this._translator(messages.sensing_current_hour);
+            case 'minute':
+                return this._translator(messages.sensing_current_minute);
+            case 'second':
+                return this._translator(messages.sensing_current_second);
+
+
+            }
```

Comment:
Nit: Couple extra newlines above

Reply:
And magically they disappear (force push)